runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java

```diff
@@ -248,6 +250,17 @@ public void syncWithRetryUnsafe(VersionLockedObject vlo,
      */
     public abstract void addTransaction(AbstractTransactionalContext tc);
 
+    /**
+     * Add an object that needs extra processing right before commit happens
+     *
+     * @param preCommitListener The context of the object that needs extra processing
+     *                         along with its lambda.
+     */
+    public abstract void addPreCommitListener(TransactionalContext.PreCommitListener preCommitListener);
+
+    @Getter
+    private List<TransactionalContext.PreCommitListener> preCommitListeners = new ArrayList<>();
```

Comment:
Please don't use TransactionalContext as a prefix. Please add TransactionalContext to imports

Reply:
done. thanks